pkg/kv/migrations/rbac_to_acl.go

```diff
@@ -0,0 +1,448 @@
+package migrations
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/treeverse/lakefs/pkg/auth"
+	"github.com/treeverse/lakefs/pkg/auth/acl"
+	"github.com/treeverse/lakefs/pkg/auth/crypt"
+	"github.com/treeverse/lakefs/pkg/auth/model"
+	authparams "github.com/treeverse/lakefs/pkg/auth/params"
+	"github.com/treeverse/lakefs/pkg/auth/wildcard"
+	"github.com/treeverse/lakefs/pkg/config"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/logging"
+	"github.com/treeverse/lakefs/pkg/permissions"
+)
+
+const (
+	maxGroups        = 1000
+	pageSize         = 1000
+	maxGroupPolicies = 1000
+)
+
+var (
+	// ErrTooMany is returned when this migration does not support a
+	// particular number of resources.  It should not occur on any
+	// reasonably-sized installation.
+	ErrTooMany         = errors.New("too many")
+	ErrTooManyPolicies = fmt.Errorf("%w policies", ErrTooMany)
+	ErrTooManyGroups   = fmt.Errorf("%w groups", ErrTooMany)
+	ErrNotAllowed      = fmt.Errorf("not allowed")
+	ErrAlreadyHasACL   = errors.New("already has ACL")
+	ErrAddedActions    = errors.New("added actions")
+	ErrEmpty           = errors.New("empty")
+	ErrPolicyExists    = errors.New("policy exists")
+
+	// allPermissions lists all permissions, from most restrictive to
+	// most permissive.  It includes "" for some edge cases.
+	allPermissions = []model.ACLPermission{"", acl.ACLRead, acl.ACLWrite, acl.ACLSuper, acl.ACLAdmin}
+)
+
+func MigrateToACL(ctx context.Context, kvStore kv.Store, cfg *config.Config, logger logging.Logger, version int, force bool) error {
+	if cfg.IsAuthTypeAPI() {
+		fmt.Println("skipping ACL migration - external Authorization")
```

Comment:
This function is called by the migrate function which checks for the return error. I assume this could be manual or automated run. Since we don't know the user's lakeFS configuration (logging) I'm not sure whether we should use prints or logs explicitly